### DISCUSSION TITLE
How to: Picture-in-Picture

### DISCUSSION THREAD

**Discussion by gkueny** [TheWidlarzGroup/react-native-video#3540] [Q&A] [answered]
### What are you trying to do?

How to implement Picture-in-Picture (PIP) so it is not tied to a specific `Video` instance.

I successfully implemented PIP in our app. But the main caveat is that PIP is tied to our `Video` instance. So, if the user navigates in our app after entering PIP mode:
- Video component is unmounted
- Video in PIP is paused
- Restoration after PIP exit never works

Do I miss something obvious that allows me to have PIP mode and a specific `Video` instance?

### What have you tried?

I tried many different approaches like:
- Have a global `Video` instance and display it in position absolute when needed. But it's very difficult to show the video exactly where we want in the app (video is shown in a scrollView and it's difficult to sync scroll position with global video position)
- Have a global `Video` instance and a specific video instance and jump between them on PIP and not PIP mode. But it's difficult to sync them both and introduces many other issues

**Comment by gkueny** [comment]
For now I implement it with a global video instance that I position with "absolute". But I'm curious to know if you have a better alternative

**Comment by KrzysztofMoch** [comment, marked as answer]
I don't think there is better solution 😕 - current architecture of library that host player instance in view isn't grate and has limits. Once we publish stable v6 and we add new arch support I will try to rewrite this lib from zero.

**Reply by gkueny** [reply]
thanks for the response @KrzysztofMoch 

I'm pretty happy with our implementation but I'm worried to miss something

**Comment by fredrikburmester** [comment]
Hi, how did you make PiP work? Could you give some code examples? Thanks

**Comment by parthkanani93** [comment]
@gkueny Can you please share the global video implimentation example i need to do that things on my app.
